content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+        });
+
+        chatChannel.subscribe("chatMessage", (data) => {
+            var dataObj = JSON.parse(JSON.stringify(data));
+            var message = dataObj.data.message;
+            var date = new Date(dataObj.data.date);
+            setState(prevState => {
+                return {
+                  ...prevState,
+                  newMsgs: [...prevState.newMsgs, {"summary": message, "image": you.avatar, "date": date, "name": you.name}]
+                    };
+                });
+        })
+
+      }, []);
+```
+
+As you can see we subscribe to many channels and the publish our information. Let's discuss each of them.
+Using the @presenceChannel@ we mark the presence of the user and also subscribe to @enter@. We also publish the @userInfo@ data to the @chatChannel@.
+Then using the @chatChannel@ we subscribe to the @userInfo@ and get user data.
+Lastly using the @chatChannel@ we subscribe to the @chatMessage@ to get the updates coming in to the newsfeed. We then update the state variables with the new update.
+
+Now, add the following imports statements on top of the file after the @Ably@ import statement to enable use of semantic-ui:
+
+```[javascript]
+import { Container, Header, Divider, Button, Form, TextArea } from 'semantic-ui-react'
+```
+
+Now, finally let’s add the @sendMyMessage@ function to the @App@ component and the HTML to post an update:
+
+```[javascript]
+function sendMyMessage() {
+        const newMsgId = "msg-id-" + Math.random().toString(36).substr(2, 6);
+        setTodoInput("");
+        if (todoInput !== "") {
+            chatChannel.publish("chatMessage", {
+                message: todoInput,
+                localMsgId: newMsgId,
+                date: new Date()
+            });
+            setState(prevState => {
+                return {
+                  ...prevState,
+                  msgs: [...prevState.msgs, {"summary": todoInput, "image": my.avatar, "date": new Date(), "name": my.name}].sort((a,b)=>b.date-a.date)
+                };
+            });
+        }
+    }
+
+    return (
+        <Container textAlign='left' className='App-header'>
+          <Header as='h2'>Ably React Tutorial - RealTime</Header>  
+          <Form>
+             <TextArea 
+                value={todoInput}
+                placeholder="Post an update"
+                onChange={e => setTodoInput(e.target.value)}
+             />
+             <div className="App-button"><Button content='Send' primary onClick={sendMyMessage} /></div>
+         </Form>
+             
+         <Divider />
+         
+         </Container>
+    )
+```
+
+As you can see in the @sendMyMessage@ function we publish the message to @chatMessage@ using @chatChannel@ and update the local state accordingly with the new update posted by the user.
+
+"See this step in Github":https://github.com/ably/tutorials
+
+h2. Step 4 - Displaying the updates in the newsfeed
+Firstly, update the following imports statements in @App.js@ file:
```

Comment:
Up until the previous step, the messaging was more like 'let's do this', 'we'll add this' etc. Please update the same for this step to maintain consistency.